script/reset-db.bash

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd -P )"
+project_dir="$script_dir"/..
+
+errmsg() {
+    IFS=' '
+    printf '%s\n' "$*" 1>&2
+}
+
+trap_exit() {
+    exit_status="$?"
+    if [ -f "$script_dir"/rackup.pid ]; then
+        kill $(<"$script_dir"/rackup.pid) || true
+    fi
+    if [ $exit_status -ne 0 ]; then
+        set +o xtrace
+        errmsg 'Something went wrong during execution.'
+        errmsg 'Exit status was: '"$exit_status"
+    fi
+}
+trap trap_exit EXIT
+
+print_help() {
+    set +o xtrace
+    errmsg 'Options:'
+    errmsg '-d <path>: Read ingest data from the file given. Mutually exclusive with -n.'
+    errmsg '-h: Print this help message.'
+    errmsg '-n: Do not pre-populate any data. Mutually exclusive with -d.'
+    set -o xtrace
+}
+
+data_file=''
+populate_data='true'
+while getopts ":d:n" opt; do
+    case $opt in
+        d)
+            data_file="$OPTARG"
+            ;;
+        h)
+            print_help
+            exit 0
+            ;;
+        n)
+            populate_data='false'
+            ;;
+        \?)
+            errmsg "Invalid option: -$OPTARG"
+            print_help
+            exit 64
+            ;;
+    esac
+done
+
+# Validate parameters
+if [ "$populate_data" != 'true' ] && [ -n "$data_file" ]; then
+    errmsg 'Error: -d and -n cannot be specified at the same time.'
+    errmsg
+    print_help
+    exit 64
+fi
+
+# This script assumes that authentication, etc. has already been set
+# up as per the system setup document referenced in the README.
+
+if pgrep -f 'localhost:9292'; then
+    set +o xtrace
+    errmsg 'There is already a running Rack instance. Please shut it down before continuing.'
+    exit 1
+fi
+
+# Many Ruby tools assume a specific working directory... even some
+# tools that claim to support a directory parameter don't work 100%
+# without changing the working directory.
```

Comment:
rake is expected to be run from where the Rakefile is, not sure we _really_ need to call this out

Reply:
There is a `--rakefile` option that takes an arbitrary filename. It works sometimes. I wasted a fair amount of time a while back learning about the "sometimes" part. There are also Ruby build and deployment tools that make faulty assumptions about the environment they are running in (assume an interactive shell, assume that if `sudo` was used it was invoked a specific way, etc.) although I don't remember exactly where. I can remove the comment, it's basically coming out of a number of bad experiences I've had trying to securely automate deployment of Ruby apps in the past and I felt like my approach is sufficiently unusual here to require some comment, just maybe not this one.

Background on the subshell: as a general policy I don't like writing scripts that depend on having a specific working directory because I think it's very hard to keep track of the last time you called `cd` in a script, especially a complex one with multiple functions etc., and I think it can make refactoring a lot harder. So in general if I need a specific working directory for a series of commands, I put the whole thing in a subshell to isolate it. (I think I might've started doing that because it also came in handy for some old Sun Java installers that would somehow export environment variables during execution and break scripts).